test/libsolidity/syntaxTests/abiEncoder/encodeCallMismatch.sol

```diff
@@ -0,0 +1,45 @@
+
+contract C {
+	function f(int a) public {}
+	function f2(int a, string memory b) public {}
+	function f3(int a, int b) public {}
+
+	function fail() public returns(bytes memory) {
+		return abi.encodeCall(this.f, ("test"));
+	}
+	function fail2() public returns(bytes memory) {
+		return abi.encodeCall(this.f, (1, 2));
+	}
+	function fail3() public returns(bytes memory) {
+		return abi.encodeCall(this.f, ());
+	}
+	function fail4() public returns(bytes memory) {
+		return abi.encodeCall(this.f);
+	}
+	function fail5() public returns(bytes memory) {
+		return abi.encodeCall(1, this.f);
+	}
+	function fail6() public returns(bytes memory) {
+		return abi.encodeCall(abi.encodeCall, (1, 2, 3, "test"));
+	}
+	function fail7() public returns(bytes memory) {
+		return abi.encodeCall(this.f3, [1, 2]);
+	}
+	function success() public returns(bytes memory) {
+		return abi.encodeCall(this.f, (1));
+	}
+	function success2() public returns(bytes memory) {
+		return abi.encodeCall(this.f, 1);
+	}
+	function success3() public returns(bytes memory) {
+		return abi.encodeCall(this.f2, (1, "test"));
+	}
```

Comment:
These also could have more meaningful names to explain what they test. Here's how I would name them:
- `fail()` -> `failFunctionArgsWrongType()`
- `fail2()` -> `failFunctionArgsTooMany()`
- `fail3()` -> `failFunctionArgsTooFew()`
- `fail4()` -> `failFunctionPtrMissing()`
- `fail5()` -> `failFunctionPtrWrongType()`
- `fail6()` -> `failFunctionPtrBuiltin()`
- `fail7()` -> `failFunctionArgsArrayLiteral()`
- `success()` -> `successFunctionArgsIntLiteralTuple()`
- `success2()` -> `successFunctionArgsIntLiteral()`
- `success3()` -> `successFunctionArgsLiteralTuple()`